nextest-runner/src/platform.rs

```diff
@@ -21,22 +23,52 @@ pub struct BuildPlatforms {
     /// In the future, this will become a list of target triples once multiple `--target` arguments
     /// are supported.
     pub target: Option<TargetTriple>,
+
+    /// The sysroot for the target platform.
+    ///
+    /// This is needed to run procedure macro binaries on Windows. See
+    /// [https://github.com/nextest-rs/nextest/issues/1493] for details. Empty if failed to detect.
+    pub sysroot: Option<Box<Utf8Path>>,
 }
 
 impl BuildPlatforms {
     /// Creates a new `BuildPlatform`.
     ///
     /// Returns an error if the host platform could not be determined.
-    pub fn new(target: Option<TargetTriple>) -> Result<Self, UnknownHostPlatform> {
+    pub fn new(
+        reader: Option<impl io::BufRead>,
+        target: Option<TargetTriple>,
+    ) -> Result<Self, UnknownHostPlatform> {
+        let sysroot = reader.and_then(|mut reader| {
+            let mut line = Default::default();
+            match reader.read_line(&mut line) {
+                Ok(_) => {
+                    let original_line = line.as_str();
+                    let line = line.trim();
+                    if line.is_empty() {
+                        log::debug!(
```

Comment:
I think this should be a warning -- not being able to find the sysroot is a bit weird.

Reply:
We've already added the warning log at [line 115 of `nextest-runner/src/list/rust_build_meta.rs`](https://github.com/nextest-rs/nextest/pull/1499/files/9c98c32f70611d86c088242b5862a6a48456b030#diff-5a11bb30b860bf0cc695f0d8140cdbe4b3d461bd8a542b335e7fe6e088476970R115) where we actually generate the `dylib_paths`. Let me know if we still want `log::warn!` here.

Reply:
ahhh I see. I think in that case, the warning is fine. Could you document this in a comment?

Reply:
Done.